cpu/nrf52/periph/spi.c

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright (C) 2014-2016 Freie Universität Berlin
+ * Copyright (C) 2020 Inria
+ * Copyright (C) 2020 Koen Zandberg <koen@bergzand.net>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_nrf52
+ * @ingroup     drivers_periph_spi
+ * @{
+ *
+ * @file
+ * @brief       Low-level SPI driver implementation based on the SPIM peripheral
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @author      Frank Holtz <frank-riot2015@holtznet.de>
+ * @author      Jan Wagner <mail@jwagner.eu>
+ * @author      Koen Zandberg <koen@bergzand.net>
+ *
+ * @}
+ */
+
+#include "cpu.h"
+#include "mutex.h"
+#include "assert.h"
+#include "periph/spi.h"
+#include "periph/gpio.h"
+#include "periph_cpu.h"
+#include <string.h>
+
+#define RAM_MASK            (0x20000000)
+
+/**
+ * @brief   array holding one pre-initialized mutex for each SPI device
+ */
+static mutex_t locks[SPI_NUMOF];
+
+/**
+ * @brief   array with a busy mutex for each SPI device, used to block the
+ *          thread until the transfer is done
+ */
+static mutex_t busy[SPI_NUMOF];
+
+static uint8_t _mbuf[SPI_NUMOF][SPI_MBUF_SIZE];
+
+static void spi_isr_handler(void *arg);
+#ifdef ERRATA_SPI_SINGLE_BYTE_WORKAROUND
+void spi_gpio_handler(void *arg);
+#endif
+
+static inline NRF_SPIM_Type *dev(spi_t bus)
+{
+    return (NRF_SPIM_Type *)spi_config[bus].dev;
+}
+
+static inline bool _in_ram(const uint8_t *data)
+{
+    return ((uint32_t)data & RAM_MASK);
+}
+
+#ifdef ERRATA_SPI_SINGLE_BYTE_WORKAROUND
+/**
+ * @brief Work-around for transmitting 1 byte with SPIM on the nrf52832.
+ * @warning Must not be used when transmitting multiple bytes.
+ * @warning After this workaround is used, the user must reset the PPI channel and the GPIOTE channel before attempting to transmit multiple bytes.
+ *
+ * @param bus   The SPI instance that is in use.
+ */
+static void _setup_workaround_for_ftpan_58(spi_t bus)
+{
+    gpio_init_int(spi_config[bus].sclk, GPIO_OUT, GPIO_BOTH,
```

Comment:
```suggestion
#ifdef ERRATA_SPI_SINGLE_BYTE_WORKAROUND
    gpio_init_int(spi_config[bus].sclk, GPIO_OUT, GPIO_BOTH,
    …
#else
    (void) bus;
#endif
```

will save yourself even more ifdefs

Reply:
Ah this is already in an ifdef block